api/graphite_proxy.go

```diff
@@ -46,5 +50,20 @@ func NewGraphiteProxy(u *url.URL) *httputil.ReverseProxy {
 		resp.Header.Del("content-length")
 		return nil
 	}
+
+	graphiteProxy.ErrorHandler = func(rw http.ResponseWriter, req *http.Request, err error) {
+		if graphiteProxy.ErrorLog != nil {
+			graphiteProxy.ErrorLog.Printf("http: proxy error: %v", err)
+		} else {
+			log.Printf("http: proxy error: %v", err)
+		}
+
+		if req.Context().Err() == context.Canceled {
+			// if the client disconnected before the query was fully processed
```

Comment:
are there other possible reasons for this to happen than the client disconnecting?

Reply:
I cannot really find any spot where the context is canceled in mt's code.

Reply:
i'm trying to find documentation showing that macaron cancels the context when the client disconnected, but i can't find anything relevant. how did you figure that out?

Reply:
> i'm trying to find documentation showing that macaron cancels the context when the client disconnected, but i can't find anything relevant. how did you figure that out?

I ran the test snippet in the bug report

Reply:
fair enough. let's merge it then